### PR TITLE
feat: add tier.run service and check if user is registered

### PR DESCRIPTION
**Pull request by brandonroberts**
## Description

- Adds service to check/register users through tier.run/Stripe for feature flag support
- Adds new environment variables example config
- Updates test to mock TierService

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/tier/tier.service.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            description: `OpenSauced Free Account`,
+            phone: "",
+            metadata: {
+              login,
+            },
+          },
+        });
+      } catch (e) {
+        this.logger.error(`Unable to register account for ${login}`);
+      }
```
Comment (jpmcb) [inline src/tier/tier.service.ts:38]: Do we want to surface the error here? How we've done error handling in the past has confused me abit due to typescripts gnarly handling of errors but elsewhere I've seen us do:

https://github.com/open-sauced/api/blob/692f953192dff8449f3716a0a28c88f20f1f741c/src/blogs/blog-summary.service.ts#L31-L34

File: src/user/services/user.service.ts [hunk 2]
```diff
@@ -261,6 +263,12 @@ export class UserService {
         });
       }
 
+      await this.tierService.checkAddOrg(
+        id,
+        { email: email as string, name: name as string, login: user_name as string },
+        user.role >= 50
```
Comment (jpmcb) [inline src/user/services/user.service.ts:269]: So "user.role greater than or equal to 50" is how we grandfather legacy privileged accounts into the tier.run system?

**Review by jpmcb — APPROVED** [review]

File: src/user/services/user.service.ts (hunk 2 above)
Comment (brandonroberts) [reply src/user/services/user.service.ts:269]: Yes, but on second thought we can do that outside of the API and leave this to the free tier.

File: src/tier/tier.service.ts (hunk 1 above)
Comment (brandonroberts) [reply src/tier/tier.service.ts:38]: Yep, good catch

**Pull request merged**